src/main/java/net/dv8tion/jda/core/requests/Requester.java

```diff
@@ -159,70 +180,41 @@ public JDAImpl getJDA()
             retryAfter = rateLimiter.handleResponse(route, response);
             if (!rays.isEmpty())
                 LOG.debug("Received response with following cf-rays: " + rays);
+
             if (retryAfter == null)
-                apiRequest.getRestAction().handleResponse(new Response(response.getStatus(), response.getBody(), -1), apiRequest);
+                apiRequest.handleResponse(new Response(response, -1, rays));
+            else if (handleOnRatelimit)
+                apiRequest.handleResponse(new Response(response, retryAfter, rays));
+            else
+                response.close();
```

Comment:
I have adjusted this to close all responses in an additional `finalize` block. For that I created a new `okhttp3.Response[4]` which holds all possible responses then closes them before exiting the method. Once one of the elements is `null` the iteration of the array is intercepted and code-flow continues.